lib/install/management/finder.go

```diff
@@ -257,49 +273,49 @@ func (d *Dispatcher) SearchVCHs(computePath string) ([]*vm.VirtualMachine, error
 	return vchs, nil
 }
 
-// searchVCHsFromComputePath searches for VCHs in all child ResourcePools under computePath.
-// The computePath can itself be a ResourcePool, ComputeResource or ClusterComputeResource.
-func (d *Dispatcher) searchVCHsFromComputePath(computePath string) ([]*vm.VirtualMachine, error) {
-	defer trace.End(trace.Begin(computePath, d.op))
-
-	pools, err := d.listResourcePools(path.Join(computePath, "..."))
-
+// search finds the compute resources based on the search path and then retrieves
+// the resouce pools for that compute resource.  Once a list of resource pools is
+// collected search will iterate over the pools and search for VCHs in each pool
+func (d *Dispatcher) search(searchPath string) ([]*vm.VirtualMachine, error) {
+	defer trace.End(trace.Begin(searchPath, d.op))
+	var vchs []*vm.VirtualMachine
+	// find compute resources for the search path
+	resources, err := d.session.Finder.ComputeResourceList(d.op, searchPath)
 	if err != nil {
-		err = errors.Errorf("Failed to search for resource pools for compute path %q: %s", computePath, err)
+		if _, ok := err.(*find.NotFoundError); ok {
+			return nil, nil
+		}
 		return nil, err
 	}
 
-	vchs := d.searchVCHsFromPools(pools)
-	return vchs, nil
-}
-
-// searchVCHsPerDC outputs a list of all VCH VMs under a datacenter
-func (d *Dispatcher) searchVCHsPerDC(dc *object.Datacenter) ([]*vm.VirtualMachine, error) {
-	defer trace.End(trace.Begin(dc.InventoryPath, d.op))
-
-	d.session.Datacenter = dc
-	d.session.Finder.SetDatacenter(dc)
-
-	pools, err := d.listResourcePools("*")
-	if err != nil {
-		err = errors.Errorf("Failed to search resource pools for datacenter %q: %s", dc.InventoryPath, err)
-		return nil, err
+	// iterate over clusters and search each resourcePool in the cluster
+	for _, c := range resources {
+		d.op.Infof("  Searching in ComputeResource %s...", c.Name())
```

Comment:
Can we shift many of these info messages to debug so that it's not being quite so verbose in large environments?

Reply:
sure...I added them so that in a larger environment a user could kill the job if they thought it was to wide a search...additionally..in a large environment it will just sit...giving no indication that it's actually doing something...